components/function-controller/cmd/webhook/main.go

```diff
@@ -53,16 +53,19 @@ func main() {
 		panic(errors.Wrap(err, "while reading env variables"))
 	}
 
-	l, err := serverlessLogging.ConfigureLogger(cfg.LogLevel, cfg.LogFormat)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	loggerRegistry, err := configureLogger(ctx, cfg.FileConfigPath)
 	if err != nil {
 		setupLog.Error(err, "unable to configure log")
 		os.Exit(1)
 	}
 
-	logrZap := zapr.NewLogger(l.WithContext().Desugar())
+	logrZap := zapr.NewLogger(loggerRegistry.CreateDesugared())
 	ctrl.SetLogger(logrZap)
 
-	log := l.WithContext()
+	log := loggerRegistry.CreateUnregistered()
```

Comment:
We rename `log`to `initLogger` as it has specific role and it shouldn't be use in components.